Objects/longobject.c

```diff
@@ -514,12 +514,11 @@ PyLong_AsLongAndOverflow(PyObject *vv, int *overflow)
             i = -(i);
         }
         while (--i >= 0) {
-            prev = x;
-            x = (x << PyLong_SHIFT) | v->ob_digit[i];
-            if ((x >> PyLong_SHIFT) != prev) {
+            if (x > (unsigned long)-1 >> PyLong_SHIFT) {
```

Comment:
It would be good to add a comment here (and other similar lines below). Something along the lines of
```
/* The right hand of this comparison if the largest unsigned long value
 * that can be shifted left by PyLong_SHIFT bits without overflow. */
```

Reply:
IMO it's quite obvious from the code.

Reply:
I would prefer ULONG_MAX rather than (unsigned long)-1. Same comment for (size_t)-1 below.